### PR TITLE
Sudo in harmonic

### PR DESCRIPTION
**Pull request by Hoverbear**
Call sudo from within harmonic instead of the trampoline

### CODE REVIEW

File: src/cli/mod.rs [hunk 1]
```diff
@@ -77,5 +79,31 @@ pub(crate) async fn signal_channel() -> eyre::Result<(Sender<()>, Receiver<()>)>
...
+        let sudo_cstring = CString::new("sudo").wrap_err("Making C string of `sudo`")?;
+        let args = std::env::args();
+        let mut arg_vec_cstring = vec![];
+        arg_vec_cstring.push(sudo_cstring.clone());
+
+        for arg in args {
+            arg_vec_cstring.push(CString::new(arg).wrap_err("Making arg into C string")?);
+        }
+
+        tracing::trace!("Execv'ing `{sudo_cstring:?}` with args `{arg_vec_cstring:?}`");
```
Comment (grahamc) [inline src/cli/mod.rs:104]:
```suggestion
        tracing::trace!("Execvp'ing `{sudo_cstring:?}` with args `{arg_vec_cstring:?}`");
```

**Review by grahamc — COMMENTED** [review]
Good improvement :+1:

**Review by grahamc — COMMENTED** [review]
Looks solid, should be good after this minor nit.

File: src/cli/mod.rs [hunk 2]
```diff
@@ -77,5 +79,53 @@ pub(crate) async fn signal_channel() -> eyre::Result<(Sender<()>, Receiver<()>)>
...
+            };
+            if preserve {
+                preserve_env_list.push(key);
+            }
+        }
+
+        arg_vec_cstring.push(
+            CString::new(format!("--preserve-env={}", preserve_env_list.join(",")))
+                .wrap_err("Building a `--preserve-env` argument for `sudo`")?,
+        );
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply src/cli/mod.rs:120]: Oh!!! Good catch.

**Pull request merged**